### PR TITLE
necessary fix for Genesis 2015-17 

### PR DESCRIPTION
**Pull request by xx979xx**
Genesis 2015-17 faults with OP because 1- "icon" only accept (2) as value 
2- if it detect steering command below 35 mph, it will hard fault and restarting is required

### CODE REVIEW

**Comment by xx979xx** [comment]
tested and it is working
However, because of the "forced disabled", SCC won't works below steering speed (35 mph)
I did some locale modification to disable "forced disabled" this this case and added auto re-engage when speed exceed steering speed. I will be happy to push it if you like the idea

File: selfdrive/car/hyundai/hyundaican.py [hunk 1]
```diff
@@ -8,7 +8,7 @@ def make_can_msg(addr, dat, alt):
 
 def create_lkas11(packer, car_fingerprint, apply_steer, steer_req, cnt, enabled, lkas11, hud_alert, keep_stock=False):
   values = {
-    "CF_Lkas_Icon": 3 if enabled else 0,
+    "CF_Lkas_Icon": 2 if car_fingerprint in FEATURES["icon_basic"] else 3 if enabled else 0,
```
Comment (rbiasini) [inline selfdrive/car/hyundai/hyundaican.py]: hard to read :) can we move the if/else outside the dict?

**Comment by rbiasini** [comment]
mmm... can SCC remain engage below 35mph? If yes, then the logic to just issue a no steer alert below 35MPH already exists:
```
https://github.com/commaai/openpilot/pull/722/files#diff-4617cfb78ac68938078ce751a8944fcdR287
```
So we just need to block steer commands being sent in carControllers below this speed.

**Review by xx979xx — COMMENTED** [review]
Sure why not
the code has moved

**Comment by xx979xx** [comment]
> mmm... can SCC remain engage below 35mph? If yes, then the logic to just issue a no steer alert below 35MPH already exists:
> https://github.com/commaai/openpilot/pull/722/files#diff-4617cfb78ac68938078ce751a8944fcdR287
> 
> So we just need to block steer commands being sent in carcontroller below this speed.

Yes SCC engaged below 35 mph. but LKAS must not be active (OP disabled)
I tested by this code and I have the same fault [https://github.com/xx979xx/openpilot/blob/18abc7ae7dfe395f806411c0aa87340997f46149/selfdrive/car/hyundai/interface.py#L288](url)

**Comment by andrewhlr** [comment]
Min speed is 60kph (37.3mph) instead of 35mph. Please update this or else the car will fault if traveling between 35 and 37.3mph for more than a second or so. Also, I’m still on 5.5 and simply have the steering requests stop below 60kph just like rbiasini recommenced. Disabled OP completely created an annoyance of having to reengage the system once the speed goes back above 60kph.

**Comment by xx979xx** [comment]
>  Also, I’m still on 5.5 and simply have the steering requests stop below 60kph just like rbiasini recommenced. Disabled OP completely created an annoyance of having to reengage the system once the speed goes back above 60kph.

I am happy to use your code, could you share it?
also, I added auto re-engaged feature in branch devel-genesis1

**Review by xx979xx — COMMENTED** [review]
speed updated

**Comment by andrewhlr** [comment]
> > Also, I’m still on 5.5 and simply have the steering requests stop below 60kph just like rbiasini recommenced. Disabled OP completely created an annoyance of having to reengage the system once the speed goes back above 60kph.
> 
> I am happy to use your code, could you share it?
> also, I added auto re-engaged feature in branch devel-genesis1

https://github.com/andrewhlr/openpilot/blob/alca-dev-055/selfdrive/car/hyundai/carcontroller.py
I added (CS.v_ego < 16.67) to line 87 but it's a quick and dirty fix. Your way may be better.

**Comment by xx979xx** [comment]
> Sure why not
> the code has moved

bad code not working as intended

**Review by xx979xx — COMMENTED** [review]
not working
`  if car_fingerprint in FEATURES["icon_basic"]:
    values["CF_Lkas_Icon"] = 2`

**Review by xx979xx — COMMENTED** [review]
this is working but speed cut it not reliable 
`if self.low_speed_alert:
events.append(create_event('belowSteerSpeed', [ET.WARNING, ET.PRE_ENABLE]))`

**Review by xx979xx — COMMENTED** [review]
this is working but I don't think @rbiasini would like it

**Pull request closed without merging**